### ISSUE TITLE
wrong month format in caption of gantt diagram

### ISSUE THREAD

**Issue by pashutkin** [plantuml/plantuml#1380]
**Describe the bug**
Wrong month formatting is used for table captions of gantt diagram. Some languages/locales use different word for standalone month name and month as a part of date string.
English word "January" is used for both standalone month name (January'70) and part of date string (The 1st of January 1970). 
But Ukrainian, Czech, Greek, Croatian, Russian, Polish, Finnish, Catalan languages have different words. For example, in Greek: "Ιανουάριος 1970" as standalone month and "1 Ιανουαρίου 1970" as a part of date string.

The problem is in [Month.java](https://github.com/plantuml/plantuml/blob/06978b87d03775b387f789bd0390f4eb532fa927/src/net/sourceforge/plantuml/project/time/Month.java#L65): TextStyle.FULL_STANDALONE and TextStyle.FULL can't be used interchangeably. Specifically it might not be decided on `longest` criteria.
I would consider to use two separate methods for standalone partial usage of month name.

**To Reproduce**

```plantuml
@startgantt
language en
project starts on 1970-01-01
[test] lasts 15 days
@endgantt
```

```plantuml
@startgantt
language el
project starts on 1970-01-01
[test] lasts 15 days
@endgantt
```

```plantuml
@startgantt
language fi
project starts on 1970-01-01
[test] lasts 15 days
@endgantt
```

```plantuml
@startgantt
language ru
project starts on 1970-01-01
[test] lasts 15 days
@endgantt
```

**Expected behavior**
Correspondingly, captions have to be:
en: `January 1970` — correct
el: `Ιανουάριος 1970` instead of `Ιανουαρίου 1970`
fi: `Tammikuu 1970` instead of `Tammikuuta 1970`
ru: `Январь 1970` instead of `Января 1970`

**Screenshots**
<img width="306" alt="image" src="https://user-images.githubusercontent.com/15768610/229517038-758ccf40-e15a-4f61-8a78-6862496d6a90.png">

**Additional context**
To illustrate the difference of month representation one may run following code in `java`:
```java
        for (var language : java.util.List.of("en", "de", "uk","cs","el","hr","ru","pl","fi","ca")) {
            var locale = java.util.Locale.forLanguageTag(language);
            String month = new java.text.SimpleDateFormat("LLLL yyyy", locale).format(new java.util.Date(0));
            String date = java.text.SimpleDateFormat.getDateInstance(java.text.DateFormat.LONG, locale).format(new java.util.Date(0));
            String fullStandaloneName = java.time.Month.JANUARY.getDisplayName(java.time.format.TextStyle.FULL_STANDALONE, locale);
            String fullName = java.time.Month.JANUARY.getDisplayName(java.time.format.TextStyle.FULL, locale);
            int comparison = fullStandaloneName.length() - fullName.length();
            System.out.printf("""
                    Locale "%1$s" / %2$s
                        FULL_STANDALONE = "%3$s", usage: "%4$s"
                        FULL = "%5$s", usage: "%6$s"
                        FULL_STANDALONE.length() - FULL.length() = %7$d
                    %n""", locale, locale.getDisplayName(), fullStandaloneName, month, fullName, date, comparison);
        }
```

**Comment by arnaudroques** [comment]
Thanks for the report and for the detailed explanations, it really helps!

The difference between `FULL_STANDALONE` and `FULL` was not clear to us :-)

Still, we are not sure to have fixed the issue in last commit.

Is this better better this way ?

https://github.com/plantuml/plantuml/blob/87caaebdb9e19e6da5bc68e000b216377b078a0f/src/net/sourceforge/plantuml/project/time/Month.java#L70

**Comment by pashutkin** [comment]
Yes. Makes sense. At least as long as the method `longName` is used only for rendering month name without 'day' part.

Thank you very much!
